velox/connectors/hive/storage_adapters/gcs/GCSFileSystem.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/file/FileSystems.h"
+
+namespace facebook::velox::filesystems {
+
+// Implementation of GCS filesystem and file interface.
+// We provide a registration method for read and write files so the appropriate
+// type of file can be constructed based on a filename. See the
+// (register|generate)ReadFile and (register|generate)WriteFile functions.
+class GCSFileSystem : public FileSystem {
```

Comment:
Can you add some documentation for the functions below ?

Reply:
I added a few lines documenting the functions of GCS used by the methods that are supported.